tests/storage/mpathconf_test.py

```diff
@@ -224,3 +224,133 @@ def test_mpathd_parser():
             Section('entry', [])
         ]),
     ]
+
+
+def test_mpathd_ufn_enabled_single():
+    fake_conf = [
+        Section('defaults', [
+            Pair('key', 'value'),
+            Pair('user_friendly_names', 'yes'),
+        ]),
+        Section('devices', [
+            Section('device', [
+                Pair('user_friendly_names', 'no'),
+                Pair('key', 'value'),
+            ]),
+            Section('device', [
+                Pair('key', 'value')
+            ])
+        ]),
+    ]
+    issues = mpathconf._check_conf(fake_conf)
+    assert issues == [
+        Section('defaults', [
+            Pair('key', 'value'),
+            Pair('user_friendly_names', 'yes'),
+        ])
+    ]
+
+
+def test_mpathd_ufn_enabled_multiple():
+    fake_conf = [
+        Section('defaults', [
+            Pair('key', 'value'),
+            Pair('user_friendly_names', 'yes'),
+        ]),
+        Section('overrides', [
+            Pair('user_friendly_names', 'yes'),
+        ]),
+        Section('devices', [
+            Section('device', [
+                Pair('user_friendly_names', 'yes'),
+                Pair('key', 'value'),
+            ]),
+            Section('device', [
+                Pair('key1', 'value1'),
+                Pair('user_friendly_names', 'yes'),
+                Pair('key2', 'value2'),
+            ]),
+            Section('device', [
+                Pair('key', 'value')
+            ])
+        ]),
+    ]
+    issues = mpathconf._check_conf(fake_conf)
+    assert issues == [
+        Section('defaults', [
+            Pair('key', 'value'),
+            Pair('user_friendly_names', 'yes'),
+        ]),
+        Section('overrides', [
+            Pair('user_friendly_names', 'yes'),
+        ]),
+        Section('device', [
+            Pair('user_friendly_names', 'yes'),
+            Pair('key', 'value'),
+        ]),
+        Section('device', [
+            Pair('key1', 'value1'),
+            Pair('user_friendly_names', 'yes'),
+            Pair('key2', 'value2'),
+        ]),
+    ]
+
+
+def test_mpathd_ufn_blacklist():
+    fake_conf = [
+        Section('defaults', [
+            Pair('key', 'value'),
+            Pair('user_friendly_names', 'no'),
+        ]),
+        Section('blacklist', [
+            Pair('key', 'value'),
+            Section('name1', [
+                Pair('key1', 'value1'),
+                Pair('key2', 'value2'),
+            ]),
+            Section('name2', [
+                Pair('key', 'value')
+            ])
```

Comment:
There is also blacklist_exceptions. We need to test both of them with the parser (currently it does not have mixed sections in the config), and in the checker.